File: event_back/src/main/java/br/thullyoo/event_back/service/UserService.java
package br.thullyoo.event_back.service;

import br.thullyoo.event_back.dto.request.user.UserRegisterRequest;
import br.thullyoo.event_back.dto.request.user.UserUpdateRequest;
import br.thullyoo.event_back.model.User;

public interface UserService {

    public User registerUser(UserRegisterRequest userRequest);

    public void updateUser(UserUpdateRequest userRequest);
}

File: event_front/src/app/services/user/user.service.ts
import { HttpClient } from '@angular/common/http';
import { inject, Injectable } from '@angular/core';
import type { UserRegisterRequest } from '../../model/user/UserRegisterRequest';
import { Router } from '@angular/router';

@Injectable({
  providedIn: 'root'
})
export class UserService {
  
  HttpClient = inject(HttpClient);

  private router = inject(Router);

  private url = "http://localhost:8080/api/user";

  registerUser(userRequest: UserRegisterRequest) {
    this.HttpClient.post(this.url + "/register", userRequest).subscribe({
      next: () =>{
        this.router.navigateByUrl("/login");
      }
      ,
      error: () =>{
        alert("Erro ao tentar fazer registro")
      }
    })
  }
}
